crypto/src/actions.rs

```diff
@@ -0,0 +1,71 @@
+use ark_ff::UniformRand;
+use rand_core::{CryptoRng, RngCore};
+
+use decaf377::{Fq, Fr};
+use decaf377_rdsa;
+
+use crate::addresses::PaymentAddress;
+use crate::keys::{EphemeralPublicKey, SpendAuthorizationKey};
+use crate::note::{Note, NoteCommitment};
+use crate::nullifier::Nullifier;
+use crate::value::{Commitment, Value};
+
+pub struct OutputDescriptionBody {
```

Comment:
On naming / code organization, since we'll eventually have a bunch of actions, I think it would be good to have this be `action::output::Body` (i.e., `actions.rs -> action.rs`, with `output` and `spend` submodules).  This way we can keep the names short and tidy and keep the code for the different actions in different submodules.

Reply:
i dig it